### PR TITLE
[CMM] Implement Ghoulish Impetus

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Ghoulish Impetus]]

File: Mage.Sets/src/mage/cards/g/GhoulishImpetus.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted creature gets +1/+1, has deathtouch, and is goaded.
+        Ability ability = new SimpleStaticAbility(new BoostEnchantedEffect(1, 1));
+        ability.addEffect(new GainAbilityAttachedEffect(DeathtouchAbility.getInstance(), AttachmentType.AURA).concatBy(", "));
+        ability.addEffect(new GoadAttachedEffect().concatBy(", and"));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GhoulishImpetus.java:42]: A reasonable assumption, but the word "and" is already present
```
     public GoadAttachedEffect() {
        super(Duration.WhileOnBattlefield, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
        staticText = "and is goaded";
    }
```

File: Mage.Sets/src/mage/cards/g/GhoulishImpetus.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        // When enchanted creature dies, return Ghoulish Impetus to the battlefield at the beginning of the next end step.
+        this.addAbility(new DiesAttachedTriggeredAbility(
+            new CreateDelayedTriggeredAbilityEffect(
+                new AtTheBeginOfNextEndStepDelayedTriggeredAbility(
+                    new ReturnToBattlefieldUnderYourControlTargetEffect()
+                        .setText("return {this} to the battlefield"),
+                    TargetController.ANY
+                )
+            ).setText("return {this} to the battlefield at the beginning of the next end step"),
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GhoulishImpetus.java:53]: I think you also have to `setTriggerPhrase("")`

File: Mage/src/main/java/mage/abilities/common/DiesAttachedTriggeredAbility.java [hunk 3]
```diff
@@ -21,6 +21,7 @@ public class DiesAttachedTriggeredAbility extends TriggeredAbilityImpl {
     private final String attachedDescription;
     private final boolean diesRuleText;
     protected SetTargetPointer setTargetPointer;
+    private final boolean rememberSource; // if true, setTargetPointer will be set to the aura/equipement.
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/DiesAttachedTriggeredAbility.java:24]: This seems like the sort of thing that ideally ought to be handled by a `SetTargetPointer` case?  Unfortunately that enum has no comments. Might be more trouble than it's worth

**Review by xenohedron — COMMENTED** [review]
Minor text issues which could always be fixed later, and a potential improvement idea that isn't necessary, depends what you think

File: Mage.Sets/src/mage/cards/g/GhoulishImpetus.java (hunk 2 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/g/GhoulishImpetus.java:53]: Don't think it is needed?

Here is current card rule (with the "and and goaded", that one you are correct), the trigger on death, and the delayed trigger at end of turn:
andand
dying trigger
returntrigger

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/g/GhoulishImpetus.java:53]: Huh okay, guess that works, I've needed that in the past for some reason but can't remember the details now.

File: Mage/src/main/java/mage/abilities/common/DiesAttachedTriggeredAbility.java (hunk 3 above)
Comment (Susucre) [reply Mage/src/main/java/mage/abilities/common/DiesAttachedTriggeredAbility.java:24]: I could introduce another SetTargetPointer enum field ATTACHMENT i guess?
PERMANENT_TARGET and ATTACHED_TO_CONTROLLER have only 2 usage each.

File: Mage.Sets/src/mage/cards/g/GhoulishImpetus.java (hunk 1 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/g/GhoulishImpetus.java]: image

**Review by xenohedron — APPROVED** [review]

**Pull request merged**